src/core/queryClient.ts

```diff
@@ -532,13 +533,35 @@ export class QueryClient {
     }
   }
 
+  findFirstMatchingQueryDefaults(
+    queryKey?: QueryKey
+  ): QueryOptions<any, any, any> | undefined {
+    if (!queryKey) {
+      return undefined
+    }
+
+    // First retrieve all matching defaults for the given key
+    const matchingDefaults = this.queryDefaults.filter(x =>
+      partialMatchKey(queryKey, x.queryKey)
+    )
+    // It is ok not having defaults, but it is error prone to have more than 1 default for a given key
+    if (matchingDefaults?.length > 1) {
```

Comment:
can `matchingDefaults` be undefined here? I don't think so as `.filter` will always return an aray:

```suggestion
    if (matchingDefaults.length > 1) {
```